common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts

```diff
@@ -0,0 +1,293 @@
+import { ComplexGestureSource } from "../../complexGestureSource.js";
+import { SimpleGestureSource, SimpleGestureSourceSubview } from "../../simpleGestureSource.js";
+
+import { GestureModel, GestureResolution, GestureResolutionSpec, RejectionDefault, ResolutionItemSpec } from "../specs/gestureModel.js";
+
+import { ManagedPromise, TimeoutPromise } from "@keymanapp/web-utils";
+import { FulfillmentCause, PathMatcher } from "./pathMatcher.js";
+
+//
+// TODO:  Next up - let's unit-test this bad boy!
+//
+
+export interface MatchResult<Type> {
+  matched: boolean,
+  action: GestureResolution<Type>
+}
+
+export interface MatchResultSpec {
+  matched: boolean,
+  action: GestureResolutionSpec
+}
+
+export class GestureMatcher<Type> {
+  private sustainTimerPromise?: TimeoutPromise;
+  public readonly model: GestureModel<Type>;
+
+  public readonly pathMatchers: PathMatcher<Type>[];
+  private predecessor?: GestureMatcher<Type>;
+
+  private readonly publishedPromise: ManagedPromise<MatchResult<Type>>; // unsure on the actual typing at the moment.
```

Comment:
Is this a TODO?

Reply:
I guess it is.  It'll settle out with the next PR or two in the sequence; they'll be focused on the parts that _consume_ this type.  

Since those parts aren't written yet, I wasn't 100% sure that the typing completely meets the gesture-selection algorithm's needs yet.  For what it's worth... it does seem to be stable thus far in my work on the consuming algorithm, but I haven't completed it yet.